### PR TITLE
feat: request command log and output

### LINKED ISSUE #957: sasjs request - enable `--log` and `--output` flags
Currently when running `sasjs request` and the SAS program does not return a valid JSON response, the following appears in the console:

```
 ERROR  An error occurred while executing the request.
 The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received undefined
```

Without a log, the above can be very difficult to debug.  We should therefore enable a `--log` (or `-l`) flag (and print the location of the log in the console output).

The same logic can also be applied to the response json (with a `-o` or `--output` flag, same as [sasjs job execute](https://cli.sasjs.io/job).

# Functional Testing

The following should work:

1. create two test services using SAS Studio, eg:

```sas
filename mc url "https://raw.githubusercontent.com/sasjs/core/main/all.sas";
%inc mc;
filename ft15f001 temp;
parmcards4;
  %webout(OPEN)
  data fish;
    set sashelp.fish;
  run;
  %webout(OBJ,fish)
  %webout(CLOSE)
;;;;
%mp_createwebservice(path=/Public/app/adapter/test,name=fish)
parmcards4;
  %abort;
;;;;
%mp_createwebservice(path=/Public/app/adapter/test,name=err)
```

2.  Execute the following command:

```bash
sasjs request /Public/app/adapter/test/fish
```

Expected:  
* No errors in console
* sasjsbuild/output.json contains data
* sasjsbuild/output.log contains log

3. Execute the following command:
```bash
sasjs request /Public/app/adapter/test/err
```

Expected:  
* Error in console
* Log provided in sasjsbuild/output.log

4. Execute the following command:
```bash
sasjs request /Public/app/adapter/test/err -l ./mylog.txt
```

Expected:  
* Error in console
* Log provided in ./mylog.txt

5.  Execute the following command:

```bash
sasjs request /Public/app/adapter/test/fish -o ./data.json
```

Expected:  
* No errors in console
* ./data.json contains data
* sasjsbuild/output.log contains log

- [x] The `-l` flag works on both SAS 9 and Viya
- [x] The `-o` flag works on both SAS 9 and Viya
- [x] Tests are added
- [x] The CLI docs are updated (both cli.sasjs.io and the terminal help)
- [ ] The `sasjs request` video is re-recorded (Task for Allan)

### PR DESCRIPTION
**Pull request by medjedovicm**
## Issue

Fixes #957

## Intent

Adding log and output arguments to the `request` command

## Implementation

- Moved `saveLog` and `geLogFilePath` functions into `utils`

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
image
- [x] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
Mocked:
image
Server:
- [x] All CI checks are green.
- [x] JSDoc comments have been added or updated.
- [x] Reviewer is assigned.

### CODE REVIEW

File: src/commands/request/spec/runRequest/err.sas [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+%mend mf_getuser; 
+/* if calling viya service with _job param, _program will conflict */
+/* so it is provided by SASjs instead as __program */
+%global __program _program;
+%let _program=%sysfunc(coalescec(&__program,&_program));
+ 
+%macro webout(action,ds,dslabel=,fmt=);
+  %mv_webout(&action,ds=&ds,dslabel=&dslabel,fmt=&fmt)
+%mend;
+  %abort;
```
Comment (allanbowe) [inline src/commands/request/spec/runRequest/err.sas:397]: actually this file only needs one line

`%abort;`

File: src/commands/request/request.ts [hunk 2]
```diff
@@ -105,29 +116,93 @@ export async function runSasJob(
...
+          logData = err.error.details.result
+        }
+
+        if (err.result) {
+          logData = err.result
+        }
+      }
+
+      if (jobPath) {
+        await saveLog(logData, logFile, jobPath, false)
```
Comment (saadjutt01) [inline src/commands/request/request.ts:146]: Can we [saveLog](https://github.com/sasjs/cli/blob/main/src/commands/flow/internal/saveLog.ts) from flow/internal.

If yes, we can extract that from internal to src/utils.

File: src/commands/request/requestCommand.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import { ServerType } from '@sasjs/utils'
 import { CommandExample, ReturnCode } from '../../types/command'
 import { TargetCommand } from '../../types/command/targetCommand'
-import { displayError, getAuthConfig } from '../../utils'
+import { displayError, getAuthConfig, prefixAppLoc } from '../../utils'
 import { runSasJob } from './request'
+import { getLogFilePath } from '../../utils/getLogFilePath'
 
 const syntax = 'request <sasProgramPath> [options]'
 const usage =
```
Comment (saadjutt01) [inline src/commands/request/requestCommand.ts:9]: Usage template also needs to be updated.

**Review by saadjutt01 — CHANGES_REQUESTED** [review]

File: src/commands/request/request.ts (hunk 2 above)
Comment (medjedovicm) [reply src/commands/request/request.ts:146]: Good point but I think that should be handled in a new issue since it's not quick fix.

**Comment by allanbowe** [comment]
I wonder if this can be helpful for requests like this: https://communities.sas.com/t5/SAS-Stored-Processes/How-to-fetch-the-log-of-a-stored-process-when-it-executes-in-the/m-p/781281/emcs_t/S2h8ZW1haWx8Ym9hcmRfc3Vic2NyaXB0aW9ufEtXNkpJQVZGNTNZUTJPfDc4MTI4MXxTVUJTQ1JJUFRJT05TfGhL#M4647

**Comment by allanbowe** [comment]
I got the following when running against a SAS 9 global target:

image

**Comment by VladislavParhomchik** [comment]
Server tests failed:
image

**Comment by VladislavParhomchik** [comment]
image

**Review by saadjutt01 — APPROVED** [review]

**Pull request merged**